AnimalMatcher/AnimalMatcher.Data/AnimalMatcherDbContext.cs

```diff
@@ -37,6 +39,20 @@ protected override void OnModelCreating(ModelBuilder builder)
                 .HasForeignKey(like => like.LikedPetId)
                 .OnDelete(DeleteBehavior.Restrict);
 
+            builder
+                .Entity<Pet>()
+                .HasMany(pet => pet.SentMessages)
+                .WithOne(message => message.Sender)
+                .HasForeignKey(message => message.RecipientId)
+                .OnDelete(DeleteBehavior.Restrict);
+
+            builder
+                .Entity<Pet>()
+                .HasMany(pet => pet.ReceivedMessages)
+                .WithOne(message => message.Recipient)
+                .HasForeignKey(message => message.SenderId)
```

Comment:
The SenderId should be a foreign key for the Sender and should be linked to SentMessages in Pet

Reply:
Good point, updated with last commit.